gap/AbelianClosure.gi

```diff
@@ -10,37 +10,65 @@ InstallMethodWithCache( AbelianClosure,
+          abelian_closure;
     
     L := FiniteCompletion( algebroid, range_category_of_hom_structure : FinalizeCategory := true );
     
     A := FreydCategory( L : FinalizeCategory := true );
     
-    object_constructor :=
-      function( abelian_closure, objA )
-        
-        return AsObjectInWrapperCategory( abelian_closure, objA );
-        
-    end;
+    ##
+    object_constructor := AsObjectInWrapperCategory;
     
-    morphism_constructor :=
-      function( abelian_closure, source, morA, range )
-        
-        return AsMorphismInWrapperCategory( abelian_closure, source, morA, range );
-        
-    end;
+    ##
+    modeling_tower_object_constructor := { cat, obj } -> obj;
+    
+    ##
+    object_datum := { cat, o } -> UnderlyingCell( o );
+    
+    ##
+    modeling_tower_object_datum := { cat, obj } -> obj;
     
-    abelian_closure := WrapperCategory( A,
-                               rec( name := Concatenation( "AbelianClosure( ", Name( algebroid ), " )" ),
-                                    object_constructor := object_constructor,
-                                    morphism_constructor := morphism_constructor,
-                                    category_filter := IsWrapperCapCategory and IsAbelianClosure,
-                                    category_object_filter := IsWrapperCapCategoryObject and IsObjectInAbelianClosure,
-                                    category_morphism_filter := IsWrapperCapCategoryMorphism and IsMorphismInAbelianClosure,
-                                    only_primitive_operations := true ) );
+    ##
+    morphism_constructor := AsMorphismInWrapperCategory;
+    
+    ##
+    modeling_tower_morphism_constructor := { cat, source, mor, range } -> mor;
+    
+    ##
+    morphism_datum := { cat, m } -> UnderlyingCell( m );
+    
+    ##
+    modeling_tower_morphism_datum := { cat, mor } -> mor;
+    
+    abelian_closure :=
+      WrapperCategory( A,
+              rec( name := Concatenation( "AbelianClosure( ", Name( algebroid ), " )" ),
+                   category_filter := IsWrapperCapCategory and IsAbelianClosure,
+                   category_object_filter := IsWrapperCapCategoryObject and IsObjectInAbelianClosure,
+                   category_morphism_filter := IsWrapperCapCategoryMorphism and IsMorphismInAbelianClosure,
+                   object_constructor := object_constructor,
+                   object_datum := object_datum,
+                   morphism_datum := morphism_datum,
+                   morphism_constructor := morphism_constructor,
+                   modeling_tower_object_constructor := modeling_tower_object_constructor,
+                   modeling_tower_object_datum := modeling_tower_object_datum,
+                   modeling_tower_morphism_constructor := modeling_tower_morphism_constructor,
+                   modeling_tower_morphism_datum := modeling_tower_morphism_datum,
+                   only_primitive_operations := true ) );
```

Comment:
If I see this correctly, you keep all values at the default and only add filters to the existing filters. Are you planning to use `WrapperCategories` in this way in the future, or is this a temporary thing and you plan to change the data structures in the future? In the first case, I can relax the checks so that you don't have to specify `(modeling_tower)_object_constructor` et al. everywhere.

Reply:
I see this as a process. I will usually start (like I did here) by the wrapper category retaining the entire modeling tower until things work (of course without compilation). But even then I need to further specialize `category_filter`, `category_object_filter`, `category_morphism_filter`. Then I would start using the wrapper category to collapse the tower (either fully or partially). There I have to specify the remaining options bundled together. So yes, I am in favor of partial relaxation.

Reply:
Okay, done in https://github.com/homalg-project/CAP_project/pull/987: The filters are now ignored in the check if they imply the default filters (which they do here).